### ISSUE TITLE
Demos are often dirty

### ISSUE THREAD

**Issue by plengauer** [plengauer/opentelemetry-bash#808]
Demos traces get regenerated automatically on every publish. ideally, the result is the same and they will not cause a new PR. However, there is some volatility. Mostly, its from ordering of spans. currently they are ordered by start time, especially in parallel scenarios or when a pipe is used, they start in parallel and the order is therefore volatile.
Maybe sorting by code line is an option to improve things, as long as we dont have loops in the demo it should still make a pretty trace.

it feels like this is unsolvable perfectly, but lets try to be as good as possible

**Comment by plengauer** [comment]
sort by `code.filepath` and `code.lineno`. this will result in a weird overall order, but since we build a tree, within a single node it should be fine

**Comment by plengauer** [comment]
the algorithm that i can think of with the most impact would be to sort by start time, unless child spans of the same parent are overlapping, then within the overlapping ones sort by end time. the reason for that is, they are overlapping anyway, so most likely a pipe (or a parallel which we can fix by using sleeps instead of echos) and they will not start consistently in the same order (they are racing) but they will end in the same, because the ending is contorlled by pipes incrementally getting closed front to back
